### PR TITLE
[FLOC-1567] Acceptance test: container with dataset

### PR DESCRIPTION
**Pull request by itamarst**
https://clusterhq.atlassian.net/browse/FLOC-1567

### CODE REVIEW

File: flocker/acceptance/test_api.py [hunk 1]
```diff
@@ -17,10 +17,32 @@
...
+    """
+    def can_connect():
+        s = socket.socket()
+        conn = s.connect_ex((host, port))
+        print "Connection status", conn
+        return False if conn else True
+
+    dl = loop_until(can_connect)
+    return dl
+
```
Comment (dwgebler) [inline flocker/acceptance/test_api.py]: Moving `verify_socket` is not strictly related to the issue or used in the "create container with dataset" test. Still, seems a small enough inline improvement that it's worth keeping in this branch.

**Comment by dwgebler** [comment]
The only thing I wonder is if we should maybe have some cleanup code added to these tests to leave the VMs / acceptance nodes in the same state they were before the test suite was run. It's fine if you run these on newly spun-up VMs each time but you can't run the `test_api` suite twice on the same node instances without failures resulting from the lack of cleanup.

Otherwise, tests pass and this looks good to merge.

File: flocker/acceptance/test_api.py [hunk 2]
```diff
@@ -17,10 +17,32 @@
...
+    :param bytes host: Host to connect to.
+    :param int port: Port to connect to.
+
+    :return Deferred: Firing when connection is possible.
+    """
+    def can_connect():
+        s = socket.socket()
+        conn = s.connect_ex((host, port))
+        print "Connection status", conn
+        return False if conn else True
```
Comment (exarkun) [inline flocker/acceptance/test_api.py]: Such a terrible way to say `return conn != 0`

**Comment by exarkun** [comment]
> It's fine if you run these on newly spun-up VMs each time but you can't run the test_api suite twice on the same node instances without failures resulting from the lack of cleanup.

This seems pretty significant.  It's already amazingly hard to run the acceptance tests.  Let's not make it even harder.

File: flocker/acceptance/test_api.py (hunk 2 above)
Comment (itamarst) [inline flocker/acceptance/test_api.py]: Just moved that code, it was already there!

**Comment by itamarst** [comment]
I added some relevant cleanup code in #1246; I'm going to open a follow up issue to do it right.

**Pull request merged**